trains.rb

```diff
@@ -0,0 +1,120 @@
+
+class Station
+  attr_reader :trains, :station_name
+  def initialize (station_name)
+    @station_name = station_name
+    @trains =[]
+  end
+
+  def arrival(train)
+    trains.push train
+  end
+
+  def list
+    pas_train=0
+    fr_train=0
+      trains.each do |train|
+        if train.train_type =='Passenger'
+          pas_train+=1
+        else
+          fr_train +=1
+        end
+      end
+      puts "There are #{pas_train} Passenger trains and #{fr_train} Freight trains at #{station_name} Station at the moment"
+  end
+
+  def show_trains
+      print "There are "
+      trains.each { |train| print " #{train.train_number}; "}
+      print "at the #{station_name} station"
+  end
+
+  def depart(train)
+    trains.delete(train)
+    train.station = nil
+  end
+end
+
+
+class Route
+  attr_accessor :stations
+
+  def initialize (start_station, terminal_station)
+    @stations=[start_station, terminal_station]
+  end
+
+ def add_station(station)
+   self.stations.insert(-2, station)
+ end
+
+ def delete_station(station)
+   if[stations.first, stations.last].include?(station)
+     puts "Not a good idea"
+   else
+     self.stations.delete(station)
+   end
+  end
+end
+
+
+class Train
+  attr_accessor :speed, :number_of_wagons, :station, :route
+  attr_reader :train_number, :train_type
+
+  def initialize (train_number, train_type, number_of_wagons, speed = 0)
+    @train_number = train_number
+    @train_type = train_type
+    @number_of_wagons = number_of_wagons
+    @speed = speed
```

Comment:
```suggestion
    @speed = 0
```
имеет смысл либо убрать скорость из аргументов и ставить ее равной нулю здесь, либо добавить проверку на то, что она не отрицательная пришла